pkg/apis/core/validation/shoot.go

```diff
@@ -1938,14 +1935,63 @@ func validateShootOperationContext(operation string, shoot *core.Shoot, fldPath
 	return allErrs
 }
 
-func validateShootHAControlPlaneUpdate(newMeta, oldMeta metav1.ObjectMeta, fldPath *field.Path) field.ErrorList {
+// ValidateShootHAControlPlaneUpdate validates the HA shoot control plane configuration
+func ValidateShootHAControlPlaneUpdate(newShoot, oldSnoot *core.Shoot) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	oldVal := oldMeta.Annotations[v1beta1constants.ShootAlphaControlPlaneHighAvailability]
+	// Setting both a HA annotation and ControlPlane Spec is not allowed
+	if bothHAAnnotationAndHASpecSet(newShoot) {
+		return append(allErrs, &field.Error{
+			Type:     field.ErrorTypeForbidden,
+			Field:    ".metadata.annotations & .spec.ControlPlane",
```

Comment:
```suggestion
			Field:    ".metadata.annotations",
```

To discourage usage of the annotation, let's only report this field as problematic.